src/extension-runners/firefox-desktop.js

```diff
@@ -0,0 +1,229 @@
+/* @flow */
+
+// Import flow types from npm dependencies.
+import type FirefoxProfile from 'firefox-profile';
+
+import {
+  MultipleExtensionsReloadError,
+  RemoteTempInstallNotSupported,
+  WebExtError,
+} from '../errors';
+import * as defaultFirefoxApp from '../firefox';
+import {
+  connectWithMaxRetries as defaultFirefoxConnector,
+} from '../firefox/remote';
+import {createLogger} from '../util/logger';
+// Import flow types from project files.
+import type {
+  FirefoxRDPResponseAddon,
+  RemoteFirefox,
+} from '../firefox/remote';
+import type {
+  ExtensionRunnerParams,
+} from './base';
+import type {
+  FirefoxPreferences,
+} from '../firefox/preferences';
+import type {
+  FirefoxInfo, // eslint-disable-line import/named
+} from '../firefox/index';
+
+export type FirefoxDesktopExtensionRunnerParams = ExtensionRunnerParams & {
+  // Firefox desktop CLI params.
+  customPrefs?: FirefoxPreferences,
+  browserConsole: boolean,
+  firefoxBinary: string,
+  preInstall: boolean,
+
+  // Firefox desktop injected dependencies.
+  firefoxApp: typeof defaultFirefoxApp,
+  firefoxClient: typeof defaultFirefoxConnector,
+};
+
+const log = createLogger(__filename);
+
+/**
+ * This module provide an ExtensionRunner subclass that manage an extension executed
+ * in a Firefox for Desktop instance.
+ */
+
+export class FirefoxDesktopExtensionRunner {
+  params: FirefoxDesktopExtensionRunnerParams;
+  reloadableExtensions: Map<string, string>;
```

Comment:
Hmm, is there a way to make it more obvious that this maps a source directory to an addon ID? A comment maybe?